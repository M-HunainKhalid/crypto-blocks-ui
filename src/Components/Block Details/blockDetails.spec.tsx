import { render, screen } from "@testing-library/react";
import { BlockDetails } from ".";

describe("Block Detail", () => {
  const props = {
    blockIndex: 11111111,
    size: 112,
    fee: 200,
    prevBlock:
      "0000000000000000000abb37d731cb7159f1de07b7071a72b0292c1461e9b845",
  };
  it("should render block detail component correctly", () => {
    const blockDetails = render(<BlockDetails {...props} />);
    expect(screen.getByText(props.blockIndex)).toBeInTheDocument();
    expect(screen.getByText(props.size)).toBeInTheDocument();
    expect(screen.getByText(props.fee)).toBeInTheDocument();
    expect(screen.getByText(props.prevBlock)).toBeInTheDocument();
    expect(blockDetails).toMatchSnapshot();
  });
});
