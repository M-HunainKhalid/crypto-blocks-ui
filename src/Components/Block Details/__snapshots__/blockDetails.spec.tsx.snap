// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block Detail should render block detail component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-1"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
          style="color: rgb(242, 242, 242); font-weight: bolder;"
        >
          Block Details
        </h3>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
            >
              <div
                class="MuiBox-root MuiBox-root-5"
              >
                <h6
                  class="MuiTypography-root makeStyles-detailName-3 MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Index:
                </h6>
                <p
                  class="MuiTypography-root makeStyles-detailValue-4 MuiTypography-body1"
                >
                  11111111
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
            >
              <div
                class="MuiBox-root MuiBox-root-6"
              >
                <h6
                  class="MuiTypography-root makeStyles-detailName-3 MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Size:
                </h6>
                <p
                  class="MuiTypography-root makeStyles-detailValue-4 MuiTypography-body1"
                >
                  112
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
          >
            <div
              class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
            >
              <div
                class="MuiBox-root MuiBox-root-7"
              >
                <h6
                  class="MuiTypography-root makeStyles-detailName-3 MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Fee:
                </h6>
                <p
                  class="MuiTypography-root makeStyles-detailValue-4 MuiTypography-body1"
                >
                  200
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
            >
              <div
                class="MuiBox-root MuiBox-root-8"
              >
                <h6
                  class="MuiTypography-root makeStyles-detailName-3 MuiTypography-h6 MuiTypography-gutterBottom"
                >
                  Previous Block:
                </h6>
                <p
                  class="MuiTypography-root makeStyles-detailValue-4 MuiTypography-body1"
                >
                  0000000000000000000abb37d731cb7159f1de07b7071a72b0292c1461e9b845
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
        style="color: rgb(242, 242, 242); font-weight: bolder;"
      >
        Block Details
      </h3>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-5"
            >
              <h6
                class="MuiTypography-root makeStyles-detailName-3 MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Index:
              </h6>
              <p
                class="MuiTypography-root makeStyles-detailValue-4 MuiTypography-body1"
              >
                11111111
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-6"
            >
              <h6
                class="MuiTypography-root makeStyles-detailName-3 MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Size:
              </h6>
              <p
                class="MuiTypography-root makeStyles-detailValue-4 MuiTypography-body1"
              >
                112
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-7"
            >
              <h6
                class="MuiTypography-root makeStyles-detailName-3 MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Fee:
              </h6>
              <p
                class="MuiTypography-root makeStyles-detailValue-4 MuiTypography-body1"
              >
                200
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-8"
            >
              <h6
                class="MuiTypography-root makeStyles-detailName-3 MuiTypography-h6 MuiTypography-gutterBottom"
              >
                Previous Block:
              </h6>
              <p
                class="MuiTypography-root makeStyles-detailValue-4 MuiTypography-body1"
              >
                0000000000000000000abb37d731cb7159f1de07b7071a72b0292c1461e9b845
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
